### PR TITLE
Major fixes

### LINKED ISSUE #225: frontsite: Patron cancel loan bug
Package version (if known): <version>

## Describe the bug
In the frontsite in the `your loans` section if we cancel a loan and we change pages, for example we go to the home page by clicking the ILS logo we will get an error 
## Steps to Reproduce
1. Go to `your loans`
2. Click on `cancel` for a loan
3. Click on ILS logo

## Expected behavior
Even if the action is not finished we should not get an error

## Screenshots (if applicable)
image

## Additional context
Related issues: #30

### PR DESCRIPTION
**Pull request by FlorianCassayre**
* serialize borrowing request responses (closes CERNDocumentServer/cds-ils#189)
* fix patron scrolling menu initial value
* improve yup validation (addresses #243)
  * propagate html required property to date picker input
  * schema errors (and more generally all errors) are displayed only after submission, instead of after value change
  * schemas for orders, items
* revert pagination pages calculation
* fix patron cancel loan page change bug (closes #225)

### CODE REVIEW

**Comment by FlorianCassayre** [comment]
Validation still needs refinement (and probably discussion).

File: src/lib/api/ill/serializers.js [hunk 1]
```diff
@@ -6,6 +6,16 @@ const LibrarySerializers = {
   },
 };
 
+const BorrowingRequestSerializers = {
+  responseSerializer: function(hit) {
+    return recordResponseSerializer(hit);
+  },
```
Comment (topless) [inline src/lib/api/ill/serializers.js:12]: I am not sure why we wrap it several times. What about the following suggestion, which will also allow us to pass more than one [arguments](https://github.com/inveniosoftware/react-invenio-app-ils/blob/df478ab328e35afc79e3df44180bdf12fb3c4545/src/lib/api/utils.js#L88). 
```suggestion
  responseSerializer: recordResponseSerializer
```

**Review by topless — APPROVED** [review]

File: src/lib/components/Pagination/Pagination.js [hunk 2]
```diff
@@ -19,7 +19,7 @@ class Pagination extends Component {
       ...extraParams
     } = this.props;
     // Maximum number of pages allowed (backend window limitation)
-    const maximumWindowPages = Math.ceil(
+    const maximumWindowPages = Math.floor(
```
Comment (ntarocco) [inline src/lib/components/Pagination/Pagination.js:22]: isn't it that this lead to a search error when clicking on the last page now?

**Review by ntarocco — APPROVED** [review]

File: src/lib/components/Pagination/Pagination.js (hunk 2 above)
Comment (kprzerwa) [reply src/lib/components/Pagination/Pagination.js:22]: no this fixes it, it was my fault

**Review by kprzerwa — APPROVED** [review]

**Pull request merged**